File: tests/integration/fixtures/typescript-project/src/processor.ts
/**
 * Processor module — calls from multiple modules.
 */
import { flatten, safeDivide, formatNumber } from './utils';
import { reportSum, reportTable, reportPowerSeries } from './reporter';

export function processBatch(pairs: [number, number][]): string[] {
  return pairs.map(([x, y]) => reportSum(x, y));
}

export function processTable(pairs: [number, number][]): string {
  return reportTable(pairs);
}

export function processAverages(values: number[]): number {
  const total = values.reduce((a, b) => a + b, 0);
  return safeDivide(total, values.length);
}

export function processPowerSeries(base: number, maxExp: number): string[] {
  const series = reportPowerSeries(base, maxExp);
  return series.map((v) => formatNumber(v));
}

export function processFlattenAndSum(nested: number[][]): number {
  const flat = flatten(nested);
  return flat.reduce((a, b) => a + b, 0);
}

File: tests/integration/fixtures/python-project/src/processor.py
"""Processor module that calls utils and reporter."""

from src.utils import flatten, safe_divide, format_number
from src.reporter import report_sum, report_table, report_power_series


def process_batch(pairs):
    """Process a batch of (x, y) pairs and return a list of sum reports."""
    return [report_sum(x, y) for x, y in pairs]


def process_table(pairs):
    """Process pairs and return a formatted table report."""
    return report_table(pairs)


def process_averages(values):
    """Compute the average of a list of numbers."""
    total = sum(values)
    return safe_divide(total, len(values))


def process_power_series(base, max_exp):
    """Return formatted power series for base up to max_exp."""
    series = report_power_series(base, max_exp)
    return [format_number(v) for v in series]


def process_flatten_and_sum(nested):
    """Flatten nested list and return sum."""
    flat = flatten(nested)
    return sum(flat)


def process_increment(values, delta):
    """Return a new list with each value incremented by delta using add."""
    from src.calculator import add as _add

    return [_add(v, delta) for v in values]


def process_combine(batch_a, batch_b):
    """Combine two batches element-wise using add."""
    from src.calculator import add as _add

    return [_add(a, b) for a, b in zip(batch_a, batch_b)]
